### PR TITLE
Update recipe for windows platform

### PR DESCRIPTION
**Pull request by marcelotrevisani**
Update recipe to also use lapack and minor improvements

### CODE REVIEW

**Comment by marcelotrevisani** [comment]
@grlee77 or @dirmeier , Could you please review that for me?

**Comment by grlee77** [comment]
Hi @marcelotrevisani. I am pretty sure OpenBLAS already includes its own built-in LAPACK. see for example xianyi/OpenBLAS#203 and the following line in the conda-forge recipe where LAPACKE is enabled:
https://github.com/conda-forge/lapack-feedstock/blob/29667bbed551e09ae26c1839684c96dbb3220fc8/recipe/build.sh#L14

Is there some LAPACK related functionality in Armadillo that doesn't seem to be working in the recipe?

**Comment by grlee77** [comment]
actually, I think prior to creation of the feedstock here I had also included lapack because I had also not realized it was built into OpenBLAS: https://github.com/conda-forge/staged-recipes/pull/2414#issuecomment-284162169

If you want to make a smaller PR with just the style improvements and adding yourself as a maintainer, I would be happy to merge that.

**Comment by marcelotrevisani** [comment]
I didn't know that regarding OpenBlas, thanks! :)
So, I will remove the ```lapack``` dependency.

I added that because I will need this package for another one which I'm creating the recipe and in their documentation they were very explicit to compile ```Armadillo``` with ```lapack``` and ```openblas```.

**Comment by marcelotrevisani** [comment]
Just to do some productive thing here for me, I will try to add ```Armadillo``` for windows platform as well.

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please rerender

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please rerender

**Comment by marcelotrevisani** [comment]
@grlee77 Could you please review that for me?

**Comment by grlee77** [comment]
This looks good, but I will wait a bit to see if there is any other feedback prior to merging.

The primary issue is that the Windows build will be missing sparse matrix support due to the missing ARPACK and SuperLU libraries while linux and OS X will have it. This could create confusion among users, but is probably better than not having a package at all on Windows. I am not aware whether conda-forge has any official policies relating to uniformity of package functionality across platforms.

The best solution would be to build ARPACK and SuperLU packages for Windows as well, but I'm not sure how difficult that might be. If we merge it as is, we can always update the recipe in the future if/when those become available.

**Comment by isuruf** [comment]
Why not use MSVC for compiling on Windows? Most conda-forge packages are built with MSVC on windows to avoid ABI issues

**Comment by marcelotrevisani** [comment]
@isuruf  It was just for convenience, because I already had this code. So, I will try to compile that using msvc

**Comment by marcelotrevisani** [comment]
@isuruf and @grlee77  is that ok?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -15,37 +15,39 @@ source:
...
-    - pkg-config
     - blas 1.1 {{ variant }}
     - openblas 0.2.20|0.2.20.*
-    - arpack
-    - superlu 5*
-    # - hdf5 1.8.17|1.8.17.*
+    - pkg-config  # [unix]
+    - arpack  # [unix]
+    - superlu 5.2  # [unix]
+    - m2w64-make  # [win]
```
Comment (isuruf) [inline recipe/meta.yaml]: Remove this

File: recipe/meta.yaml [hunk 2]
```diff
@@ -15,37 +15,39 @@ source:
...
-  skip: true  # [win]
+  number: 202
+  skip: true  # [win32]
   features:
     - blas_{{ variant }}
 
 requirements:
   build:
     - cmake
+    - toolchain
```
Comment (isuruf) [inline recipe/meta.yaml]: Use `{{ compiler('cxx') }}`

File: recipe/bld.bat [hunk 3]
```diff
@@ -0,0 +1,9 @@
+mkdir build
+cd build
+
+REM Trick to avoid CMake/sh.exe error
+ren "C:\Program Files\Git\usr\bin\sh.exe" _sh.exe
+
+cmake -DCMAKE_INSTALL_PREFIX="%PREFIX%" -DCMAKE_INSTALL_LIBDIR="%LIBRARY_LIB%" -DCMAKE_INSTALL_INCLUDEDIR="%LIBRARY_INC%" -DCMAKE_INSTALL_BINDIR="%LIBRARY_BIN%" -DCMAKE_INSTALL_DATADIR="%LIBRARY_PREFIX%" ..
```
Comment (isuruf) [inline recipe/bld.bat]: Add `cmake -G "%CMAKE_GENERATOR%"`

File: recipe/bld.bat [hunk 4]
```diff
@@ -0,0 +1,9 @@
+mkdir build
+cd build
+
+REM Trick to avoid CMake/sh.exe error
+ren "C:\Program Files\Git\usr\bin\sh.exe" _sh.exe
```
Comment (isuruf) [inline recipe/bld.bat]: Remove this

File: recipe/bld.bat (hunk 3 above)
Comment (isuruf) [reply recipe/bld.bat]: Also break this line into multiple lines with `^`

File: recipe/meta.yaml [hunk 5]
```diff
@@ -15,37 +15,39 @@ source:
...
     # optional items below here
-    - pkg-config
     - blas 1.1 {{ variant }}
     - openblas 0.2.20|0.2.20.*
```
Comment (isuruf) [inline recipe/meta.yaml]: Remove the version pinning here.

File: recipe/meta.yaml [hunk 6]
```diff
@@ -15,37 +15,39 @@ source:
...
-    - arpack
-    - superlu 5*
-    # - hdf5 1.8.17|1.8.17.*
+    - pkg-config  # [unix]
+    - arpack  # [unix]
+    - superlu 5.2  # [unix]
+    - m2w64-make  # [win]
   run:
     - blas 1.1 {{ variant }}
     - openblas 0.2.20|0.2.20.*
```
Comment (isuruf) [inline recipe/meta.yaml]: Remove the version pinning here

File: recipe/meta.yaml (hunk 2 above)
Comment (marcelotrevisani) [reply recipe/meta.yaml]: I testes before with ```{{ compiler('c') }}``` and ```{{ compiler('cxx') }}``` but it didn't work. However I will try it again
Do you think I have to replace ```toolchain``` by ```{{ compiler('c') }}``` ?

File: recipe/bld.bat (hunk 3 above)
Comment (marcelotrevisani) [reply recipe/bld.bat]: Sure! I'm going to change that! thanks!

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender

**Comment by marcelotrevisani** [comment]
Is it ready to be merged?

**Comment by grlee77** [comment]
LGTM.  thanks @marcelotrevisani and @isuruf

**Pull request merged**